src/test_workflow/benchmark_test/benchmark_test_cluster.py

```diff
@@ -0,0 +1,156 @@
+        self.is_endpoint_public = False
+        self.cluster_endpoint = None
+        self.cluster_endpoint_with_port = None
+        self.stack_name = f"opensearch-infra-stack-{self.args.stack_suffix}-{self.manifest.build.id}-{self.manifest.build.architecture}"
+
+    def start(self) -> None:
+        # os.chdir(self.work_dir)
+        command = f"npm install && cdk deploy \"*\" {self.params} --outputs-file {self.output_file}"
+
+        logging.info(f'Executing "{command}" in {os.getcwd()}')
+        subprocess.check_call(command, cwd=os.getcwd(), shell=True)
+        with open(self.output_file, "r") as read_file:
+            load_output = json.load(read_file)
+            self.create_endpoint(load_output)
+        self.wait_for_processing()
+
+    def create_endpoint(self, cdk_output: dict) -> None:
+        loadbalancer_url = cdk_output[self.stack_name].get('loadbalancerurl', None)
+        if loadbalancer_url is None:
+            raise RuntimeError("Unable to fetch the cluster endpoint from cdk output")
+        self.cluster_endpoint = loadbalancer_url
+        self.cluster_endpoint_with_port = "".join([loadbalancer_url, ":", str(self.port)])
+
+    @property
+    def endpoint(self) -> str:
+        return self.cluster_endpoint
+
+    @property
+    def endpoint_with_port(self) -> str:
+        return self.cluster_endpoint_with_port
+
+    @property
+    def port(self) -> int:
+        return 80 if self.args.insecure else 443
+
+    def terminate(self) -> None:
+        command = f"cdk destroy {self.stack_name} {self.params} --force"
+        logging.info(f'Executing "{command}" in {os.getcwd()}')
+
+        subprocess.check_call(command, cwd=os.getcwd(), shell=True)
+
+    def wait_for_processing(self, tries: int = 3, delay: int = 15, backoff: int = 2) -> None:
+        logging.info(f"Waiting for domain at {self.endpoint} to be up")
+        protocol = "http://" if self.args.insecure else "https://"
+        url = "".join([protocol, self.endpoint, "/_cluster/health"])
+        request_args = {"url": url} if self.args.insecure else {"url": url, "auth": HTTPBasicAuth("admin", "admin"), "verify": False}  # type: ignore
+        retry_call(requests.get, fkwargs=request_args,
+                   tries=tries, delay=delay, backoff=backoff)
+
+    def setup_cdk_params(self, config: dict) -> dict:
+        if self.args.stack_suffix:
+            suffix = self.args.stack_suffix + '-' + self.manifest.build.id + '-' + self.manifest.build.architecture
+        else:
+            suffix = self.manifest.build.id + '-' + self.manifest.build.architecture
+        return {
+            "distributionUrl": self.manifest.build.location,
+            "vpcId": config["Constants"]["VpcId"],
+            "account": config["Constants"]["AccountId"],
+            "region": config["Constants"]["Region"],
+            "suffix": suffix,
+            "securityDisabled": "true" if self.args.insecure else "false",
```

Comment:
Should be able to do str(self.args.insecure) and always get the right value of "true" or "false" - default value for insecure should be `false`

Reply:
I guess I resorted to this because the boolean in python is `True` or `False` but in typescript and many other languages it is `true` or `false`. I think better way to handle this would be:
`str(self.args.insecure).lower()`.